app/lib/notification_categories.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class NotificationCategories
+
+  AVAILABLE_CATEGORIES = %i[account billing application service service_plan alert report].freeze
+
+  def initialize(user)
+    @user = user
+  end
+
+  def enabled_categories
+    AVAILABLE_CATEGORIES.select { enabled?(_1) }
+  end
+
+  def enabled?(category)
+    case category
+    when :billing
+      @user.has_permission?(:finance) && provider_is_billing? && !provider.master_on_premises?
+    when :account
+      @user.has_permission?(:partners)
+    when :application
+      @user.has_permission?(:partners) && at_least_one_service?
+    when :alert
+      at_least_one_service? && @user.has_permission?(:monitoring)
+    when :service_plan
+      @user.account.settings.service_plans_ui_visible? &&
+        at_least_one_service? && @user.has_permission?(:partners)
+    when :report
+      @user.has_permission?(:partners) && @user.admin? && Rails.application.config.three_scale.daily_weekly_reports_pref
+    when :service
+      at_least_one_service? && @user.has_permission?(:partners)
+    else
+      false
+    end
+  end
+
+  private
+
+  def provider_is_billing?
+    provider.is_billing_buyers?
+  end
+
+  def provider
+    @provider ||= @user.account.provider? ? @user.account : @user.account.provider_account
```

Comment:
```suggestion
    account = @user.account
    @provider ||= account.provider? ? account : account.provider_account
```

but not necessary, I guess it's exactly the same in terms of efficiency.

Reply:
I'd prefer to not touch this. I don't see a benefit on setting the value of a new instance variable at every call to `provider` and then not use it from anywhere else. If any I would use a local variable instead.

Reply:
Yeah, I don't think there0s much benefit in performance to be worth the change.

Reply:
I committed the suggestion, apparently QLTY is happy now.